src/openforms/analytics_tools/tests/test_models.py

```diff
@@ -0,0 +1,69 @@
+from django.test import TestCase, override_settings
+
+from cookie_consent.models import Cookie, CookieGroup
+
+from ..constants import AnalyticsTools
+from ..models import AnalyticsToolsConfiguration, CSPSetting
+from ..utils import get_cookies, get_csp
+
+
+@override_settings(
+    SOLO_CACHE=None,
+)
+class AdminTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        cls.json_cookies = get_cookies(AnalyticsTools.google_analytics)
+        cls.json_csp = get_csp(AnalyticsTools.google_analytics)
+
+    def setUp(self):
+        super().setUp()
+        self.config = AnalyticsToolsConfiguration.get_solo()
+        self.config.analytics_cookie_consent_group = CookieGroup.objects.get(
+            varname="analytical"
+        )
+        self.config.gtm_code = "GTM-XXXX"
+        self.config.ga_code = "UA-XXXXX-Y"
+        self.config.save()
+        self.addCleanup(lambda: AnalyticsToolsConfiguration.get_solo().delete())
+
+    def test_analytics_tool_properly_enabled(self):
+
+        self.config.enable_google_analytics = True
+        self.config.save()
+
+        for cookie in self.json_cookies:
+            self.assertIsNotNone(Cookie.objects.get(name=cookie["name"]))
+        for csp in self.json_csp:
+            self.assertIsNotNone(CSPSetting.objects.get(value=csp["value"]))
+
+    def test_analytics_tool_properly_disabled(self):
+
+        # creation of cookies
+        self.config.enable_google_analytics = True
+        self.config.save()
+
+        # Deletion of cookies
+        self.config.enable_google_analytics = False
+        self.config.save()
+
+        for cookie in self.json_cookies:
+            with self.assertRaises(Cookie.DoesNotExist):
+                Cookie.objects.get(name=cookie["name"])
```

Comment:
maybe cleaner is to use `self.assertFalse(Cookie.objects.filter(name=cookie["name"]).exists())`